swift/apple/FirezoneKit/Sources/FirezoneKit/Views/UpdateNotification.swift

```diff
@@ -0,0 +1,245 @@
+      return SemanticVersion(major: parsed.major, minor: parsed.minor, patch: parsed.patch)
+  }
+
+  func encode(to encoder: Encoder) throws {
+      var container = encoder.singleValueContainer()
+      let versionString = "\(major).\(minor).\(patch)"
+      try container.encode(versionString)
+  }
+
+  static func < (lhs: SemanticVersion, rhs: SemanticVersion) -> Bool {
+      if lhs.major != rhs.major {
+          return lhs.major < rhs.major
+      }
+
+      if lhs.minor != rhs.minor {
+          return lhs.minor < rhs.minor
+      }
+
+      return lhs.patch < rhs.patch
+  }
+
+  static func == (lhs: SemanticVersion, rhs: SemanticVersion) -> Bool {
+      return lhs.major == rhs.major &&
+             lhs.minor == rhs.minor &&
+             lhs.patch == rhs.patch
+  }
+}
+
+private struct VersionInfo: Decodable {
+  let apple: SemanticVersion
+}
+
+private class UpdateNotifier: NSObject, UNUserNotificationCenterDelegate {
+  @Published private(set) var decision: UNAuthorizationStatus
+  private var lastNotifiedVersion: SemanticVersion?
+  static let lastDismissedVersionKey = "lastDismissedVersion"
+  static let notificationIdentifier = "UPDATE_CATEGORY"
+  static let appStoreLink = URL(string: "https://apps.apple.com/us/app/firezone/id6443661826")!
+
+  override public init() {
+    self.decision = .notDetermined
+    super.init()
+
+    let notificationCenter = UNUserNotificationCenter.current()
+
+    // Define a custom action
+    let dismissAction = UNNotificationAction(identifier: "DISMISS_ACTION",
+                                             title: "Dismiss This Version",
+                                             options: [])
+
+    // Define a notification category with the action
+    let notificationCategory = UNNotificationCategory(identifier: UpdateNotifier.notificationIdentifier,
+                                                       actions: [dismissAction],
+                                                       intentIdentifiers: [],
+                                                       options: [])
+
+    notificationCenter.setNotificationCategories([notificationCategory])
+
+    notificationCenter.delegate = self
+    notificationCenter.requestAuthorization(options: [.sound, .badge, .alert]) { granted, error in
+      notificationCenter.getNotificationSettings { notificationSettings in
+        self.decision = notificationSettings.authorizationStatus
+      }
+    }
+
+  }
+
+  public func updateNotification(version: SemanticVersion) {
+    if let lastDismissedVersion = getLastDismissedVersion(), lastDismissedVersion >= version {
+      return
+    }
+
+    let content = UNMutableNotificationContent()
+    lastNotifiedVersion = version
+    content.title = "Update Firezone"
+    content.body = "New Firezone version available"
+    content.sound = .default
+    content.categoryIdentifier = UpdateNotifier.notificationIdentifier
+
+    let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
```

Comment:
Here too, I think we can inline a few variables if we don't need them for anything else.